nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/WaitNotifyProtocol.java

```diff
@@ -144,6 +150,28 @@ public void setReleasableCount(int releasableCount) {
 
     }
 
+    /**
+     * Class to pass around the signal with the JSON that was deserialized to create it.
+     */
+    public static class SignalHolder {
```

Comment:
Agreed, we can definitely get right of the SignalHolder if you use the approach above.